test/test_repolib.py

```diff
@@ -1283,36 +1282,33 @@ class TestYumPluginManager(unittest.TestCase):
     ORIGINAL_DNF_PLUGIN_DIR = YumPluginManager.DNF_PLUGIN_DIR
     ORIGINAL_PLUGINS = YumPluginManager.PLUGINS
 
-    def init_dnf_plugin_conf_files(self, conf_string):
+    def tearDown(self) -> None:
+        YumPluginManager.DNF_PLUGIN_DIR = self.ORIGINAL_DNF_PLUGIN_DIR
+        YumPluginManager.PLUGINS = self.ORIGINAL_PLUGINS
+
+        self.temp_yum_dir = None
+        self.file_01 = None
+        self.file_02 = None
+
+    def init_dnf_plugin_conf_files(self, conf_string: str):
         """
         Mock configuration files of plugins
         """
-        dnf_tmp_dir = tempfile.mkdtemp()
-        f, plug_file_name_01 = tempfile.mkstemp(prefix="", suffix=".conf", dir=dnf_tmp_dir, text=True)
-        f, plug_file_name_02 = tempfile.mkstemp(prefix="", suffix=".conf", dir=dnf_tmp_dir, text=True)
-        with open(plug_file_name_01, "w") as plug_file_01:
-            plug_file_01.write(conf_string)
-        with open(plug_file_name_02, "w") as plug_file_02:
-            plug_file_02.write(conf_string)
-        self.plug_file_name_01 = plug_file_name_01
-        self.plug_file_name_02 = plug_file_name_02
-        self.tmp_dir = dnf_tmp_dir
-        YumPluginManager.DNF_PLUGIN_DIR = dnf_tmp_dir
+        self.temp_yum_dir = tempfile.TemporaryDirectory()
+        self.file_01 = tempfile.NamedTemporaryFile(suffix=".conf", dir=self.temp_yum_dir.name, delete=False)
+        self.file_02 = tempfile.NamedTemporaryFile(suffix=".conf", dir=self.temp_yum_dir.name, delete=False)
+
+        with open(self.file_01.name, "w") as handle:
+            handle.write(conf_string)
+        with open(self.file_02.name, "w") as handle:
+            handle.write(conf_string)
```

Comment:
TBH here I'd follow the same pattern done in `TestAttachCommand.setUpClass()`:
- keep using `tempfile.mkstemp()`
- write directly to the returned open fd's, closing them afterwards